packages/google_sign_in/google_sign_in_web/lib/src/flexible_size_html_element_view.dart

```diff
@@ -0,0 +1,140 @@
+
+  /// The initial Size for the widget, before it starts tracking its contents.
+  final Size? initialSize;
+
+  @override
+  State<StatefulWidget> createState() => _FlexHtmlElementView();
+}
+
+class _FlexHtmlElementView extends State<FlexHtmlElementView> {
+  /// The last measured size of the watched element.
+  Size? _lastReportedSize;
+
+  /// Watches for changes being made to the DOM tree.
+  ///
+  /// See: https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver
+  DomMutationObserver? _mutationObserver;
+
+  /// Reports changes to the dimensions of an Element's content box.
+  ///
+  /// See: https://developer.mozilla.org/en-US/docs/Web/API/Resize_Observer_API
+  DomResizeObserver? _resizeObserver;
+
+  @override
+  void dispose() {
+    // Disconnect the observers
+    _mutationObserver?.disconnect();
+    _resizeObserver?.disconnect();
+    super.dispose();
+  }
+
+  /// Update the state with the new `size`, if needed.
+  void _doResize(Size size) {
+    if (size != _lastReportedSize) {
+      domConsole.debug(
+          'Resizing', <Object>[widget.viewType, size.width, size.height]);
+      setState(() {
+        _lastReportedSize = size;
+      });
+    }
+  }
+
+  /// The function called whenever an observed resize occurs.
+  void _onResizeEntries(
+    List<DomResizeObserverEntry> resizes,
+    DomResizeObserver observer,
+  ) {
+    final DomRectReadOnly rect = resizes.last.contentRect;
+    if (rect.width > 0 && rect.height > 0) {
+      _doResize(Size(rect.width, rect.height));
+    }
+  }
+
+  /// A function which will be called on each DOM change that qualifies given the observed node and options.
+  ///
+  /// When mutations are received, this function attaches a Resize Observer to
+  /// the first child of the mutation, which will drive
+  void _onMutationRecords(
+    List<DomMutationRecord> mutations,
+    DomMutationObserver observer,
+  ) {
+    for (final DomMutationRecord mutation in mutations) {
+      if (mutation.addedNodes != null) {
+        final DomElement? element = _locateSizeProvider(mutation.addedNodes!);
+        if (element != null) {
+          _resizeObserver = createDomResizeObserver(_onResizeEntries);
+          _resizeObserver?.observe(element);
+          // Stop looking at other mutations
+          observer.disconnect();
+          return;
+        }
+      }
+    }
+  }
+
+  /// Registers a MutationObserver on the root element of the HtmlElementView.
+  void _registerListeners(DomElement? root) {
+    assert(root != null, 'DOM is not ready for the FlexHtmlElementView');
+    _mutationObserver = createDomMutationObserver(_onMutationRecords);
+    // Monitor the size of the child element, whenever it's created...
+    _mutationObserver!.observe(
+      root!,
+      childList: true,
+    ); //subtree: true);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox.fromSize(
+      size: _lastReportedSize ?? widget.initialSize ?? const Size(1, 1),
```

Comment:
FYI: using `BoxConstraints` from a `LayoutBuilder` may provide a better first guess than `Size(1, 1)`.

Reply:
For the best user experience in this usage, the `initialSize` should be guesstimated off of the button configuration.

The layout builder however tells me the constraints that flutter will apply to my widget, but it does not tell me what's the rendered size of the contents of the html element view, so I'm not sure that the estimation is going to be better (if I understand the layout builder correctly, in that case the widget will start large, and then likely shrink, and with Size(1,1) it'll start small and likely grow?)